base/loading.jl

```diff
@@ -213,11 +215,29 @@ function hash(pkg::PkgId, h::UInt)
     h += 0xc9f248583a0ca36c % UInt
     h = hash(pkg.uuid, h)
     h = hash(pkg.name, h)
+    return h
 end
 
 show(io::IO, pkg::PkgId) =
     print(io, pkg.name, " [", pkg.uuid === nothing ? "top-level" : pkg.uuid, "]")
 
+function binpack(pkg::PkgId)
+    io = IOBuffer()
+    write(io, UInt8(0))
+    uuid = pkg.uuid
+    write(io, uuid === nothing ? UInt128(0) : UInt128(uuid))
```

Comment:
We could just use `UUID(UInt128(0))` as a sentinel value instead of `nothing` to avoid the union here.